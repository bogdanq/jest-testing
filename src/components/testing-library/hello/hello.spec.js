import React from "react";
import { render } from "@testing-library/react";

import { Hello } from "./index";

describe("Hello component", () => {
  it('should render without "name" props', () => {
    const { getByText } = render(<Hello />);

    expect(getByText("Hey, stranger")).toBeDefined();
  });

  it('should render with "name" props', () => {
    const { getByText } = render(<Hello name="Bogdan" />);

    expect(getByText("Hello, Bogdan!")).toBeDefined();
  });
});
