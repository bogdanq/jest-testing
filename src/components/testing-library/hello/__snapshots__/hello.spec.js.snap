// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello component renders a message 1`] = `
<h1
  data-testid="h1"
>
  Hello, 
  Bogdan
  !
</h1>
`;
